addons/docs/src/blocks/mdx.tsx

```diff
@@ -0,0 +1,14 @@
+import * as React from 'react';
+import { DocsContext, DocsContextProps } from './DocsContext';
+
+// Hacky utility for dealing with functions or values in MDX Story elements
+export const makeStoryFn = (val: any) => (typeof val === 'function' ? val : () => val);
+
+// Hacky utilty for adding mdxStoryToId to the default context
+export const AddContext: React.FC<DocsContextProps> = props => {
```

Comment:
- possibly allow for class component as well: `export const AddContext: React.ComponentType` ?